pyensembl/database.py

```diff
@@ -83,33 +83,42 @@ def _create_database(self):
             table_name="ensembl",
             df=df,
             subdir=CACHE_SUBDIR,
-            overwrite=False,
+            overwrite=force,
             indices=indices)
         return db
 
-    def _connect_or_create_database(self):
+    def _connect_if_exists(self):
         """
-        If database already exists, open a connection.
-        Otherwise, create it.
+        Return the connection if the DB exists, and otherwise return 
+        None. As a side effect, stores the database connection in 
+        self._connection.
         """
-        db_path = self.local_db_path()
-        if exists(db_path):
-            db = sqlite3.connect(db_path)
-            # maybe file got created but not filled
-            if datacache.db.db_table_exists(db, 'ensembl'):
-                return db
-        return self._create_database()
+        if self._connection is None:
+            db_path = self.local_db_path()
+            if exists(db_path):
+                connection = sqlite3.connect(db_path)
+                # maybe file got created but not filled
+                if datacache.db.db_table_exists(connection, 'ensembl'):
+                    self._connection = connection
+        return self._connection
 
     @property
     def connection(self):
         """
         Return the sqlite3 database for this Ensembl release
-        (download and/or construct it necessary).
-        As a side effect, stores the database connection in self._db
+        (download and/or construct it if necessary, if auto_download
+        is on). As a side effect, stores the database connection in 
+        self._connection.
         """
-        if self._connection is None:
-            self._connection = self._connect_or_create_database()
-        return self._connection
+        if self._connect_if_exists():
+            return self._connection
+        if self.gtf.auto_download:
```

Comment:
Hmm. This error is raised if there's no connection to the DB *and* auto download is off.

There's a similar code path in the GTF object (see `local_gtf_path`), and the reason I have them both is that I wanted to raise an error if (a) there's no cached GTF file and (b) if there's no DB.

(I might have a GTF file without a DB, and don't want to auto-create the DB. But, if I just have the DB code path, it auto-downloads on force `index`.)

For now, I added an `auto_download` in `database.py` so I don't have to reach into GTF. Maybe `auto_download` is confusing, since it covers both downloading and indexing, but `auto_install` seems more confusing.

Let me know if you have more thoughts, though. I might be misunderstanding your suggestion.